### PR TITLE
Update CHANGE_LOG for 0.49

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,9 +1,255 @@
-Version 0.49.0dev
+Version 0.50.0dev
```
Comment (sklam) [inline CHANGE_LOG]: I think we should defer mentioning 0.50.0dev. We should do this when we tag 0.50.0dev on the master branch.

Comment (sklam) [reply CHANGE_LOG]: also, so that the release branch will have 0.49.0 as the first entry in the change log

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,9 +1,255 @@
...
+important changes to solve a number of long standing issues. In addition some
+new features have been added and a large number of bugs have been fixed!
+
+Highlights of core feature changes include:
+
+* Removal of all Python 2 related code and also updating the minimum supported
+  Python version to 3.6, the minimum supported NumPy version to 1.15 and the
+  minimum supported SciPy version to 1.0. (Stuart Archibald).
+* Refactoring of the Numba code base. The code is now organised into submodules
+  by functionality, this cleaning Numba's top level namespace
```
Comment (sklam) [inline CHANGE_LOG]: > this cleaning Numba's top level namespace

extra? runon?

```suggestion
  by functionality. This cleans up Numba's top level namespace.
```

File: CHANGE_LOG (hunk 1 above)
Comment (stuartarchibald) [reply CHANGE_LOG]: fair point, fixed!

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,9 +1,255 @@
...
+  (Pearu Peterson).
+
+Enhancements from user contributed PRs (with thanks!):
+
+* Aaron Meurer added support for thread masking at runtime in #4615.
+* Andreas Sodeur fixed a long standing bug that was preventing ``cProfile`` from
+  working with Numba JIT compiled functions in #4476.
+* Arik Funke fixed error messages in ``test_array_reductions`` (#5278) and fixed
+  an issue with test discovery (#5239).
+* Antonio Russo Fixed a SyntaxWarning in #5252.
```
Comment (sklam) [inline CHANGE_LOG]:
```suggestion
* Antonio Russo fixed a SyntaxWarning in #5252.
```

**Pull request merged**